### PR TITLE
[bitnami/moodle] Added HPA support for Moodle pods

### PR DESCRIPTION
**Pull request by SharkyND**
### Description of the change

This PR introduces horizontal pod autoscaling support for Moodle™ based on given metrics. It allows for automatic adjustment of Moodle™ replicas count depending on the specified CPU and memory utilization percentages, as well as custom metrics. Enabling this feature will override the replicaCount value while disabling it will use the fixed replicaCount parameter to set the number of Moodle™ replicas. This will make sure that we still have support replicaCount.

### Benefits

With this change, users can now enable horizontal pod autoscaling for their Moodle™ deployments, ensuring better resource management and performance under varying workloads. Think about a scenario where we are using ArgoCD or GitOps with HPA, we have to disable the static replicaCount, otherwise, HPA will not be able to scale up as ArgoCD will try to maintain this.

### Possible drawbacks

Enabling autoscaling may cause the number of replicas to fluctuate, making it more challenging to predict resource usage and costs.

### Applicable issues

N/A

### Additional information

N/A

### Checklist

- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [X] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami/readme-generator-for-helm)
- [X] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [X] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

**Review by juan131 — COMMENTED** [review]
Thanks for this great contribution @SharkyND !

One concern I have: should we add a warning in case autoscaling is enabled about RWX PVC(s)? Please note they'll be required since the same Persistent Volume will be shared among all the Moodle replicas.

File: bitnami/moodle/templates/deployment.yaml [hunk 1]
```diff
@@ -19,7 +19,9 @@ spec:
   {{- if .Values.updateStrategy }}
   strategy: {{- toYaml .Values.updateStrategy | nindent 4 }}
   {{- end }}
+  {{- if and .Values.replicaCount (not .Values.autoscaling.enabled) }}
```
[the remark this answers is not in the document]
Comment (SharkyND) [reply bitnami/moodle/templates/deployment.yaml:22]: Thanks for reviewing this @juan131. You are correct. I will add that warning to `values.yaml`

[the remark this answers is not in the document]
Comment (SharkyND) [reply bitnami/moodle/templates/deployment.yaml]: Done with the changes suggested. Thanks.

**Review by juan131 — APPROVED** [review]
LGTM! Thanks so much for this awesome contribution

**Pull request merged**